### PR TITLE
Document //flutter/runtime/dart_isolate.h

### PR DESCRIPTION
**Pull request by chinmaygarde**
Related to https://github.com/flutter/flutter/issues/42778

### CODE REVIEW

File: runtime/dart_isolate.h [hunk 1]
```diff
@@ -21,25 +21,170 @@
...
+  ///                                         isolate instance.
+  /// @param[in]  isolate_create_callback     The isolate create callback. This
+  ///                                         will be called when the before the
+  ///                                         main Dart entrypoint is invoked in
+  ///                                         the root isolate. The isolate is
+  ///                                         already in the running state at
+  ///                                         this point and an isolate scope is
+  ///                                         current.
+  /// @param[in]  isolate_shutdown_callback   The isolate shutdown callback.
+  ///                                         This will be called before it
```
Comment (jason-simmons) [inline runtime/dart_isolate.h]: typo: "it" -> "the"

File: runtime/dart_isolate.h [hunk 2]
```diff
@@ -54,55 +199,215 @@ class DartIsolate : public UIDartState {
...
+  /// @brief      Get the settings used to create this isolate instance.
+  ///
+  /// @return     The settings used in the `CreateRootIsolate` call.
+  ///
   const Settings& GetSettings() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      The current phase of the isolate. The engine represents all
+  ///             dart isolates as being in one of the known phases. By invoking
+  ///             various methods on the Dart isolate, the engine transition the
```
Comment (jason-simmons) [inline runtime/dart_isolate.h]: typo: "transition" -> "transitions"

File: runtime/dart_isolate.h [hunk 3]
```diff
@@ -54,55 +199,215 @@ class DartIsolate : public UIDartState {
...
+  ///             Dart isolate from one phase to the next. The Dart isolate will
+  ///             only move from one phase to the next in the order specified in
+  ///             the `DartIsolate::Phase` enum. That is, the once the isolate
```
Comment (jason-simmons) [inline runtime/dart_isolate.h]: typo: remove "the"

File: runtime/dart_isolate.h [hunk 4]
```diff
@@ -54,55 +199,215 @@ class DartIsolate : public UIDartState {
...
+  ///             specified library. The isolate must already be in the
+  ///             `Phase::Ready` phase.
+  ///
+  /// @param[in]  library_name  The name of the library in which to invoke the
+  ///                           supplied entrypoint.
+  /// @param[in]  entrypoint    The entrypoint in `library_name`
+  /// @param[in]  args          A list of string arguments to the entrypoint.
+  /// @param[in]  on_run        A callback to run in isolate scope after the
+  ///                           main entrypoint has been invoked. There is no
+  ///                           isolate scoped current on the thread once this
```
Comment (jason-simmons) [inline runtime/dart_isolate.h]: typo: "scoped" -> "scope"

File: runtime/dart_isolate.h [hunk 5]
```diff
@@ -54,55 +199,215 @@ class DartIsolate : public UIDartState {
...
+  /// @brief      Transition the root isolate to the `Phase::Running` phase and
+  ///             invoke the main entrypoint (the "main" method) in the root
+  ///             library. The isolate must already be in the `Phase::Ready`
+  ///             phase.
+  ///
+  /// @param[in]  entrypoint  The entrypoint in the root library.
+  /// @param[in]  args        A list of string arguments to the entrypoint.
+  /// @param[in]  on_run      A callback to run in isolate scope after the main
+  ///                         entrypoint has been invoked. There is no isolate
+  ///                         scoped current on the thread once this method
```
Comment (jason-simmons) [inline runtime/dart_isolate.h]: typo: "scoped" -> "scope"

File: runtime/dart_isolate.h (hunk 1 above)
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

File: runtime/dart_isolate.h [hunk 6]
```diff
@@ -21,25 +21,170 @@
...
+  ///                                         current.
+  /// @param[in]  isolate_shutdown_callback   The isolate shutdown callback.
+  ///                                         This will be called before it
+  ///                                         isolate is about to transition
+  ///                                         into the Shutdown phase. The
+  ///                                         isolate is still running at this
+  ///                                         point and an isolate scope is
+  ///                                         current.
+  ///
+  /// @return     { description_of_the_return_value }
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

File: runtime/dart_isolate.h (hunk 2 above)
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

File: runtime/dart_isolate.h (hunk 3 above)
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

File: runtime/dart_isolate.h (hunk 5 above)
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

File: runtime/dart_isolate.h (hunk 4 above)
Comment (chinmaygarde) [reply runtime/dart_isolate.h]: Done.

**Pull request merged**